wormhole-connect/src/hooks/useAvailableRoutes.ts

```diff
@@ -0,0 +1,89 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useDebounce } from 'use-debounce';
+
+import { RouteState, setRoutes } from 'store/transferInput';
+import RouteOperator from 'routes/operator';
+import config from 'config';
+
+import type { Route } from 'config/types';
+import type { RootState } from 'store';
+
+const useAvailableRoutes = (): void => {
+  const dispatch = useDispatch();
+
+  const { token, destToken, fromChain, toChain, amount } = useSelector(
+    (state: RootState) => state.transferInput,
+  );
+
+  const { toNativeToken } = useSelector((state: RootState) => state.relay);
+
+  const [debouncedAmount] = useDebounce(amount, 500);
+
+  useEffect(() => {
+    if (!fromChain || !toChain || !token || !destToken) {
+      return;
+    }
+
+    let isActive = true;
+
+    const getAvailable = async () => {
+      const routes: RouteState[] = [];
+      for (const value of config.routes) {
+        const r = value as Route;
+
+        let available = false;
+        let supported = false;
+
+        try {
```

Comment:
I think we should call `isRouteSupported` first and only call `isRouteAvailable` if it's supported to save on RPC calls. Also if a route is not supported, then it will not be available.